.github/workflows/publish-docs-preview.yml

```diff
@@ -11,15 +11,40 @@ jobs:
     if: "${{ github.event.label.name == ':eyes: publish-docs-preview' }}"
     steps:
 
+      - name: Find CI workflow run for PR
+        id: find-workflow-run
+        uses: actions/github-script@v7
+        continue-on-error: true
+        with:
+          script: |
+            // Find the last successful workflow run for the current PR's head
+            const {owner, repo} = context.repo;
+            const runsResponse = await github.rest.actions.listWorkflowRuns({
+              owner,
+              repo,
+              workflow_id: 'ci.yml',
+              event: 'pull_request',
+              head_sha: context.sha,
+              status: 'success'
+            });
+            const runs = runsResponse.data.workflow_runs;
+            runs.sort((a, b) => b.created_at.getTime() - a.created_at.getTime());

```

Comment:
Well, as far as I can tell, there isn't really a way to compare dates directly in JS. I guess I could compare the numbers and return -1/0/1 depending on the comparison, but it would be much more verbose...

Reply:
And that's the main reason I didn't even make my original comment!